### PR TITLE
fix(authenticator): migrate totpSecretCode generation to state machine

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Migrate `Authenticator` TOTP secret code generation from UI components to state machine:
- remove race condition causing transient errors in [`sign-in-totp-mfa`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/e2e/features/ui/components/authenticator/sign-in-totp-mfa.feature) integ test by ensuring that `Auth.setupTOTP` is only called once
- fix issue with multiple QR code images getting rendered in React `Authenticator`
- align Vue `setupTOTP` UX with other `Authenticator` implementations
- expose `totpSecretCode` in `useAuthenticator` (code is generated only for `setupTOTP` route, value is `null` for all other routes)
- prevent `getQRFields` from being called for non-`setupTOTP` routes in `useAuthenticator`
- remove extraneous usage of `useMemo` in `useAuthenticator`
- remove unit tests making assertions against TOTP secret code generation from the framework packages

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested, integ tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by calebpollman** [comment]
Moving back in to draft for the time being

**Review by ErikCH — APPROVED** [review]
I looked over the Vue changes and that will work. The `useAuthenticator` composable is a little tricky to add Types too. That could certainly be something that we can add in the future.

File: packages/react-core/src/Authenticator/hooks/useAuthenticator/useAuthenticator.ts [hunk 1]
```diff
@@ -38,38 +38,32 @@ export default function useAuthenticator(
...
   const facade = useSelector(service, xstateSelector, comparator);
 
-  const { route, unverifiedContactMethods, user, ...rest } = facade;
+  const { route, totpSecretCode, unverifiedContactMethods, user, ...rest } =
+    facade;
 
   // do not memoize output. `service.getSnapshot` reference remains stable preventing
   // `fields` from updating with current form state on value changes
-  const serviceSnapshot = service.getSnapshot();
+  const serviceSnapshot = service.getSnapshot() as AuthMachineState;
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-core/src/Authenticator/hooks/useAuthenticator/useAuthenticator.ts:46]: Discussed offline

**Review by wlee221 — APPROVED** [review]
:shipit:

**Review by ErikCH — APPROVED** [review]

**Pull request merged**